src/problems/hanoi.clj

```diff
@@ -0,0 +1,35 @@
+(ns problems.hanoi
+  (require [clojure.set :as set]))
+
+;;scardycat.clj was a practice file; this file is (hanoi.clj) our final submission
+
+(defrecord State [towers])
+
+(defn legal? [towers old-t new-t]
+  (and (not (empty? (nth towers old-t)))
+       (if (empty? (nth towers new-t))
+         true
+         (< (peek (nth towers old-t)) (peek (nth towers new-t))))))
+
+(defn swap [towers old-t new-t]
+  (let [disk (peek (nth towers old-t))
+        new-old (pop (nth towers old-t))
+        new-new (conj (nth towers new-t) disk)]
+    (assoc (assoc towers old-t new-old) new-t new-new)))
+(def towersinside [[5 4 3 2 1] [] []])
```

Comment:
These two `def`s would make more sense in some test code than cluttering up this namespace.